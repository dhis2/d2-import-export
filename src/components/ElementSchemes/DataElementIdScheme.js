import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import i18n from '@dhis2/d2-i18n';

import { optionPropType } from '../../utils/options';
import { SchemeContext } from '../../contexts/';
import { Select } from '../Select';

const DataElementIdScheme = ({ selected, setSelected, dataTest }) => {
    const { DataElementId } = useContext(SchemeContext);
    return (
        <Select
            name="dataElementIdScheme"
            label={i18n.t('Data element ID scheme')}
            options={DataElementId}
            selected={selected}
            setValue={setSelected}
            dense
            dataTest={dataTest}
        />
    );
};

DataElementIdScheme.propTypes = {
    dataTest: PropTypes.string.isRequired,
    selected: optionPropType.isRequired,
    setSelected: PropTypes.func.isRequired,
};

export { DataElementIdScheme };
