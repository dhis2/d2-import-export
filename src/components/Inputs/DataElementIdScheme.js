import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { DataElementIdScheme as DataElementIdSchemeGeneric } from '../'

const NAME = 'dataElementIdScheme'
const DATATEST = 'input-data-element-id-scheme'
const LABEL = i18n.t('Data element ID scheme')

const DataElementIdScheme = () => (
    <DataElementIdSchemeGeneric name={NAME} label={LABEL} dataTest={DATATEST} />
)

export { DataElementIdScheme }
