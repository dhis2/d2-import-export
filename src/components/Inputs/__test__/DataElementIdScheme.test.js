import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { DataElementIdScheme } from '../DataElementIdScheme'

jest.mock('react-final-form')

describe('Input component - DataElementIdScheme', () => {
    it('should render correctly', () => {
        const file = shallow(<DataElementIdScheme />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
