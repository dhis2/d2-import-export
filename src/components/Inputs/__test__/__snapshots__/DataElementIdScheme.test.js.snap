// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - DataElementIdScheme should render correctly 1`] = `
<Select
  data-test-id="input-data-element-id-scheme"
  label="Data element id scheme"
  name="dataElementIdScheme"
  options={
    Array [
      Object {
        "label": "Uid",
        "value": "UID",
      },
      Object {
        "label": "Code",
        "value": "CODE",
      },
      Object {
        "label": "Name",
        "value": "NAME",
      },
    ]
  }
/>
`;
